ophyd/device.py

```diff
@@ -86,10 +86,6 @@
 DEFAULT_CONNECTION_TIMEOUT = object()
 
 
-class OrderedDictType(Dict[A, B]):
```

Comment:
I _think_ this was here for compatibility with Python < 3.8
IIRC, it was the source of most of the type issues.
Since 3.8 is the minimum target now, I think we can just replace it with `TypedDict`